lib/galaxy/datatypes/text.py

```diff
@@ -726,3 +727,38 @@ def sniff_prefix(self, file_prefix):
         False
         """
         return file_prefix.startswith("IQ-TREE")
+
+
+@build_sniff_from_prefix
+class Yaml(Text):
+    """Yaml files"""
+    file_ext = "yaml"
+
+    def sniff_prefix(self, file_prefix):
+        """
+            Try to load the string with the yaml module. If successful it's a yaml file.
+        """
+        return self._looks_like_yaml(file_prefix)
+
+    def get_mime(self):
+        """Returns the mime type of the datatype"""
+        return 'application/yaml'
+
+    def _looks_like_yaml(self, file_prefix):
+        # Pattern used by SequenceSplitLocations
+        if file_prefix.file_size < 50000 and not file_prefix.truncated:
+            # If the file is small enough - don't guess just check.
+            try:
+                item = yaml.load(file_prefix.contents_header)
+                # exclude simple types, must set format in these cases
+                assert isinstance(item, (list, dict))
+                return True
+            except Exception:
+                return False
+        else:
+            start = file_prefix.string_io().read(100).strip()
+            if start:
```

Comment:
I feel like this will include too many non-YAML files. I'd rather you just return False but I understand that has downsides.

Reply:
I had thought of that, and my rationale was that it had to be labeled as a yaml in the filename first to trigger the sniffer. If it is bad after that, it will break and throw an error, but yaml are too open in their structure to do much else. I can't get to the computer that has this on it right now, but if you would prefer, I'll remake the pr tomorrow morning and return false if the file is too big.

Reply:
Some random remarks from the top of my head:

- I guess there are loads of test files in the Galaxy sources that we could try. 
- Wouldn't it be sufficient to check for "^---" is the first non comment line?

Reply:
- Question is also how fast the yaml reader will find out that it is not a yaml file. My guess is that there will be only a few cases where the whole file is read. Maybe we can give the while prefix/file to the loader and it will fail fast in almost all cases. 

Btw. is the file/prefix that is given to the parser complete lines or can there be truncated lines?

Reply:
The prefix is based on Galaxy's build from prefix function called at the top, and I had understood it to be full lines. I don't think, especially with the shape of yaml files, that full lines are a problem here, and in fact truncating them could cause big problems.
In regards to the other comment, looking at some files online, I had a bit of a misunderstanding as the the purpose of the "---". Since it doesn't always start the document, just separating directives, I didn't think to use it, but the case where it won't happen in the first couple hundred lines would be rare, and it would be valid as a sniff. I'll switch it out.